python3/httplib2/__init__.py

```diff
@@ -858,6 +958,112 @@ def __init__(self, host, port=None, key_file=None, cert_file=None,
                 cert_file=cert_file, timeout=timeout, context=context,
                 check_hostname=disable_ssl_certificate_validation ^ True)
 
+    def _GetValidHostsForCert(self, cert):
```

Comment:
- (important) this ought to be redundant with modern ssl module that does proper SNI check, this code was in py2 tree to support 2.6-, AFAIK or please explain how it is needed ssl module host verification is not enough here
- (minor but still) https://tldrlegal.com/license/apache-license-2.0-(apache-2.0) read section `Must`, this is not Android, we don't copy code and strip license information